src/bin/pg_autoctl/pgctl.c

```diff
@@ -272,23 +313,19 @@ ensure_default_settings_file_exists(const char *configFilePath,
 	for (settingIndex = 0; settings[settingIndex].name != NULL; settingIndex++)
 	{
 		GUC *setting = &settings[settingIndex];
+		char *value = get_current_GUC_value(settings, setting->name, pgSetup);
```

Comment:
we already know the setting that we got from settings collections. Passing the collection to for another search inside get_current_GUC_value looked unnecessary.
I think we can use
```c
get_current_GUC_value(setting, pgSetup)
```

Reply:
Fixed now, thanks!